cli/mender-artifact/main.go

```diff
@@ -0,0 +1,236 @@
+		key,
+	}
+
+	//
+	// read
+	//
+	readCommand := cli.Command{
+		Name:        "read",
+		Usage:       "Reads artifact file.",
+		ArgsUsage:   "<artifact path>",
+		Action:      readArtifact,
+		Description: "This command validates artifact file provided by pathspec.",
+		Flags:       []cli.Flag{key},
+	}
+
+	//
+	// sign
+	//
+	sign := cli.Command{
+
+		Name:        "sign",
+		Usage:       "Signs existing artifact file.",
+		Action:      signExisting,
+		UsageText:   "mender-artifact sign [options] <pathspec>",
+		Description: "This command signs artifact file provided by pathspec.",
+	}
+	sign.Flags = []cli.Flag{
+		key,
+		cli.StringFlag{
+			Name: "output-path, o",
+			Usage: "Full path to output signed artifact file; " +
+				"if none is provided existing artifact will be replaced with signed one",
+		},
+		cli.BoolFlag{
+			Name:  "force, f",
+			Usage: "Force creating new signature if the artifact is already signed",
+		},
+	}
+
+	//
+	// modify existing
+	//
+	modify := cli.Command{
+		Name:        "modify",
+		Usage:       "Modifies image or artifact file.",
+		Action:      modifyArtifact,
+		UsageText:   "mender-artifact modify [options] <pathspec>",
+		Description: "This command modifies existing image or artifact file provided by pathspec.",
+	}
+
+	modify.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "key, k",
+			Usage: "Full path to the private key that will be used to sign the artifact after modifying.",
+		},
+		cli.StringFlag{
+			Name:  "server-uri, u",
+			Usage: "Mender server URI; the default URI will be replaced with given one.",
+		},
+		cli.StringFlag{
+			Name: "server-cert, c",
+			Usage: "Full path to the certificate file that will be used for validating " +
+				"Mender server by the client.",
+		},
+		cli.StringFlag{
+			Name: "verification-key, v",
+			Usage: "Full path to the public verification key that is used by the client  " +
+				"to verify the artifact.",
+		},
+		cli.StringFlag{
+			Name:  "name, n",
+			Usage: "New name of the artifact.",
+		},
+		cli.StringFlag{
+			Name:  "tenant-token, t",
+			Usage: "Full path to the tenant token that will be injected into modified file.",
+		},
+	}
+
+	copy := cli.Command{
+		Name:        "cp",
+		Usage:       "cp <src> <dst>",
+		Description: "Copies a file into or out of a mender artifact, or sdimg",
+		UsageText:   "Copy from or into an artifact, or sdimg where either the <src> or <dst> has to be of the form [artifact|sdimg]:<filepath>",
+		Action:      Copy,
+	}
+
+	cat := cli.Command{
+		Name:        "cat",
+		Usage:       "cat [artifact|sdimg]:<filepath>",
+		Description: "Cat can output a file from a mender artifact or mender image to stdout.",
+		Action:      Cat,
+	}
+
+	append := cli.Command{
+		Name:        "append",
+		Usage:       "append [artifact|sdimg]:<filepath> <src>",
+		Description: "append or prepend to  file on a mender artifact or sdimg, where the placement of the '-' decides the append or prepend functionality",
+		Action:      Append,
+	}
```

Comment:
Should mention `uefiimg` in all the usage strings as well.

Reply:
Probably. I’ll fix it up tomorrow

Reply:
We will, see my earlier comment.